src/RestSharp/Request/RestRequestExtensions.cs

```diff
@@ -387,6 +388,23 @@ public static RestRequest AddXmlBody<T>(this RestRequest request, T obj, string
         return request;
     }
 
+    /// <summary>
+    /// Adds cookie to the <seealso cref="HttpClient"/> cookie container.
+    /// </summary>
+    /// <param name="request">RestRequest to add the cookies to</param>
+    /// <param name="name">Cookie name</param>
+    /// <param name="value">Cookie value</param>
+    /// <param name="path">Cookie path</param>
+    /// <param name="domain">Cookie domain, must not be an empty string</param>
+    /// <returns></returns>
+    public static RestRequest AddCookie(this RestRequest request, string name, string value, string path, string domain) {
+        if (request.CookieContainer == null) {
```

Comment:
We can create a cookie container on the first call of `AddCookie`

Reply:
Yeah good point. I will change it.